src/ucm/util/sys.c

```diff
@@ -235,14 +235,14 @@ void ucm_parse_proc_self_maps(ucm_proc_maps_cb_t cb, void *arg)
         /* address           perms offset   dev   inode   pathname
          * 00400000-0040b000 r-xp  00001a00 0a:0b 12345   /dev/mydev
          */
-        *newline = '\0';
+        length = newline - ptr;
         ret = sscanf(ptr, "%lx-%lx %4c %*x %*x:%*x %*d %n",
```

Comment:
shall we also add \n at the end of sscanf line, and verify that ret is exactly the match?

Reply:
with extended mutex scope we ain't gonna need it anyway